File: palindrome.java
import java.util.Scanner;
class Palindrome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter string: ");
        String str = sc.nextLine();
        String rev = new StringBuilder(str).reverse().toString();
        if (str.equals(rev)) System.out.println("Palindrome");
        else System.out.println("Not Palindrome");
    }
}

File: palindrome.c
#include <stdio.h>
int main() {
    int n, rev = 0, temp, rem;
    printf("Enter number: ");
    scanf("%d", &n);
    temp = n;
    while (n != 0) {
        rem = n % 10;
        rev = rev * 10 + rem;
        n /= 10;
    }
    if (rev == temp) printf("Palindrome\n");
    else printf("Not Palindrome\n");
    return 0;
}
